src/main/java/com/contentful/java/cma/ModuleRoles.java

```diff
@@ -0,0 +1,282 @@
+package com.contentful.java.cma;
+
+import com.contentful.java.cma.RxExtensions.DefFunc;
+import com.contentful.java.cma.model.CMAArray;
+import com.contentful.java.cma.model.CMARole;
+import com.contentful.java.cma.model.CMASystem;
+
+import java.util.concurrent.Executor;
+
+import retrofit2.Response;
+import retrofit2.Retrofit;
+
+/**
+ * This module contains all the role options available to this SDK.
+ */
+public class ModuleRoles extends AbsModule<ServiceRoles> {
+  private final Async async;
+
+  /**
+   * Create a new role module.
+   *
+   * @param retrofit
+   * @param callbackExecutor
+   */
+  public ModuleRoles(Retrofit retrofit, Executor callbackExecutor) {
+    super(retrofit, callbackExecutor);
+    async = new Async();
+  }
+
+  /**
+   * Initialize retrofit with the current service.
+   *
+   * @param retrofit the instance to use the service with.
+   * @return a populated instance of the service to be used.
+   */
+  @Override protected ServiceRoles createService(Retrofit retrofit) {
+    return retrofit.create(ServiceRoles.class);
+  }
+
+  /**
+   * Use all functionalities of this module asynchronously.
+   *
+   * @return an asynchronous handling instance.
+   */
+  public Async async() {
+    return async;
+  }
+
+  /**
+   * Fetch all roles of this space.
+   *
+   * @param spaceId the space identifier identifying the space.
+   * @return the array of roles.
+   * @throws IllegalArgumentException if spaceId is null.
+   */
+  public CMAArray<CMARole> fetchAll(String spaceId) {
+    assertNotNull(spaceId, "spaceId");
+    return service.fetchAll(spaceId).toBlocking().first();
```

Comment:
The `first()` item of this queue is actually the array, so yes, it is the _first array from contentful_. If the result is null, the server responded with an error, and it would have been an error somwhere before, which the user will either see as a crash (sync) or as an error in `onFailure` on async.